src/Core/Checkout/Order/Aggregate/OrderDelivery/OrderDeliveryDefinition.php

```diff
@@ -65,7 +66,7 @@ protected function defineFields(): FieldCollection
             (new FkField('state_id', 'stateId', StateMachineStateDefinition::class))->addFlags(new Required()),
             new ManyToOneAssociationField('stateMachineState', 'state_id', StateMachineStateDefinition::class, 'id', false),
 
-            (new StringField('tracking_code', 'trackingCode'))->addFlags(new SearchRanking(SearchRanking::HIGH_SEARCH_RANKING)),
+            (new JsonField('tracking_codes', 'trackingCodes'))->addFlags(new Required(), new SearchRanking(SearchRanking::HIGH_SEARCH_RANKING)),
```

Comment:
```suggestion
            (new ListField('tracking_codes', 'trackingCodes', StringField::class))->addFlags(new Required(), new SearchRanking(SearchRanking::HIGH_SEARCH_RANKING)),
```

That way the DAL only accepts json arrays and no objects and checks the values inside the array to be strings